### PR TITLE
fixes #10574

### LINKED ISSUE #10574: [🐛 Bug]: Intercepting network response produces exception
### What happened?

Intercepting network response crashes the browser. 

@titusfortner [said](https://stackoverflow.com/a/71975826/2620206): 
the code is assuming responseHeaders will be there, but that's optional according to [chromedevtools.github.io/devtools-protocol/tot/Fetch](https://chromedevtools.github.io/devtools-protocol/tot/Fetch/)

### How can we reproduce the issue?

```shell
#issue is reproduced by typing `irb` and pasting the code below

require "watir"
require "selenium/devtools"

Selenium::WebDriver::Chrome::Service.driver_path = '/chromedriver99.0.4844.51'
Selenium::WebDriver::Chrome.path = '/chrome99.0.4844.84/google-chrome'
options = Selenium::WebDriver::Chrome::Options.new
options.add_option("args", ["--no-sandbox"])

browser = Watir::Browser.new :chrome, options: options

log_file = File.open("/tmp/network.log", 'w')
browser.wd.intercept do |request, &continue|
    continue.call(request) do |response|
      log_file << "#{request.id} \t
                   #{request.method} \t
                   #{response.code} \t
                   #{request.url} \n"
    end
end

browser.goto "youtube.com" #this crashes the browser
```

### Relevant log output

```shell
#<Thread:0x00005590f1bcda90@/selenium-webdriver-4.1.0/lib/selenium/webdriver/devtools.rb:145 run> terminated with exception (report_on_exception is true):
Traceback (most recent call last):
        3: from /selenium-webdriver-4.1.0/lib/selenium/webdriver/devtools.rb:155:in `block in callback_thread'
        2: from /selenium-webdriver-4.1.0/lib/selenium/webdriver/common/driver_extensions/has_network_interception.rb:68:in `block in intercept'
        1: from /selenium-webdriver-4.1.0/lib/selenium/webdriver/common/driver_extensions/has_network_interception.rb:109:in `intercept_response'
/selenium-webdriver-4.1.0/lib/selenium/webdriver/devtools/response.rb:38:in `from': undefined method `each_with_object' for nil:NilClass (NoMethodError)
```

### Operating System

Ubuntu

### Selenium version

ruby 2.6.5p114 (2019-10-01 revision 67812) [x86_64-linux]

### What are the browser(s) and version(s) where you see this issue?

Chrome 99.0.4844.84

### What are the browser driver(s) and version(s) where you see this issue?

ChromeDriver99.0.4844.51

### Are you using Selenium Grid?

No

### PR DESCRIPTION
**Pull request by meetme2meat**
Attempt to fix https://github.com/SeleniumHQ/selenium/issues/10574

### CODE REVIEW

File: rb/lib/selenium/webdriver/devtools/response.rb [hunk 1]
```diff
@@ -35,7 +35,7 @@ def self.from(id, encoded_body, params)
             id: id,
             code: params['responseStatusCode'],
             body: (Base64.strict_decode64(encoded_body) if encoded_body),
-            headers: params['responseHeaders'].each_with_object({}) do |header, hash|
+            headers: (params['responseHeaders'] ||  []).each_with_object({}) do |header, hash|
```
Comment (titusfortner) [inline rb/lib/selenium/webdriver/devtools/response.rb]: Interestingly, if we'd used `#fetch` here we would have had a better error message. That was your initial suggestion, any reason for the change? We probably should be moving in that direction for our Hash implementations in general (tough to enforce that in a linter)

**Review by titusfortner — CHANGES_REQUESTED** [review]
you can run rubocop from the `rb/` directory to see what our settings flag on.

File: rb/lib/selenium/webdriver/devtools/response.rb (hunk 1 above)
Comment (meetme2meat) [reply rb/lib/selenium/webdriver/devtools/response.rb]: Done. Zero Rubocop warning and used `params.fetch`

**Review by titusfortner — APPROVED** [review]

File: rb/lib/selenium/webdriver/devtools/response.rb (hunk 1 above)
Comment (titusfortner) [reply rb/lib/selenium/webdriver/devtools/response.rb]: Thanks, sorry for delay

**Pull request merged**